openslides/motions/projector.py

```diff
@@ -334,10 +335,10 @@ async def motion_block_slide(
     # All title information for referenced motions in the recommendation
     referenced_motions: Dict[int, Dict[str, str]] = {}
 
-    # Search motions.
-    all_motions = await all_data_provider.get_collection("motions/motion")
-    for motion in all_motions.values():
-        if motion["motion_block_id"] == motion_block["id"]:
+    # iterate motions.
+    for motion_id in motion_block["motions_id"]:
+        motion = await all_data_provider.get("motions/motion", motion_id)
+        if motion:
```

Comment:
Same here. THis if-clause shoudl not be necessary. If there isn't the motion, something is not updated correctly in the cache.